File: backend/models/Dependent.js
const mongoose = require('mongoose');

const dependentSchema = new mongoose.Schema({
	patient: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
	name: { type: String, required: true, trim: true, maxlength: 100 },
	relationship: { type: String, required: true, trim: true, maxlength: 40 },
	dateOfBirth: { type: Date, required: true },
	gender: { type: String, enum: ['male', 'female', 'other', 'prefer_not_to_say'] },
	phone: { type: String, trim: true, maxlength: 20 }
}, { timestamps: true });

module.exports = mongoose.model('Dependent', dependentSchema);

File: lib/models/dependent.dart
class Dependent {
  const Dependent({
    required this.id,
    required this.name,
    required this.relationship,
    required this.dateOfBirth,
    this.gender,
    this.phone,
  });

  final String id;
  final String name;
  final String relationship;
  final DateTime dateOfBirth;
  final String? gender;
  final String? phone;

  factory Dependent.fromJson(Map<String, dynamic> json) => Dependent(
    id: json['id']?.toString() ?? json['_id']?.toString() ?? '',
    name: json['name'] as String? ?? '',
    relationship: json['relationship'] as String? ?? '',
    dateOfBirth: DateTime.parse(json['dateOfBirth'].toString()),
    gender: json['gender'] as String?,
    phone: json['phone'] as String?,
  );

  Map<String, dynamic> toJson() => {
    'name': name,
    'relationship': relationship,
    'dateOfBirth': dateOfBirth.toIso8601String(),
    if (gender != null) 'gender': gender,
    if (phone != null) 'phone': phone,
  };
}
